src/main/java/org/brapi/test/BrAPITestServer/service/core/BatchService.java

```diff
@@ -0,0 +1,194 @@
+
+	public BatchService(BatchRepository batchRepository, PeopleService peopleService) {
+		this.batchRepository = batchRepository;
+		this.peopleService = peopleService;
+	}
+
+	public BatchDetails getBatch(String batchDbId) throws BrAPIServerException {
+		BatchEntity entity;
+
+		Optional<BatchEntity> entityOpt = batchRepository.findById(batchDbId);
+		if (entityOpt.isPresent()) {
+			entity = entityOpt.get();
+		} else {
+			throw new BrAPIServerDbIdNotFoundException("batch", batchDbId, HttpStatus.NOT_FOUND);
+		}
+
+		return convertToDetails(entity);
+	}
+
+	public BatchDetails updateBatchItems(String batchDbId, List<String> batchItems) throws BrAPIServerException {
+		BatchEntity savedEntity;
+		Optional<BatchEntity> entityOpt = batchRepository.findById(batchDbId);
+		if (entityOpt.isPresent()) {
+			BatchEntity entity = entityOpt.get();
+			entity.setDateModified(new Date());
+
+			List<BatchItemEntity> itemEntities = batchItems.stream().map((item) -> {
+				BatchItemEntity itemEntity = new BatchItemEntity();
+				itemEntity.setItem(item);
+				itemEntity.setBatch(entity);
+				return itemEntity;
+			}).collect(Collectors.toList());
+
+			entity.getData().addAll(itemEntities);
+
+			savedEntity = batchRepository.save(entity);
+		} else {
+			throw new BrAPIServerDbIdNotFoundException("batch", batchDbId, HttpStatus.NOT_FOUND);
+		}
+
+		return convertToDetails(savedEntity);
+	}
+
+	public BatchDetails updateBatch(String batchDbId, BatchNewRequest batch) throws BrAPIServerException {
+		BatchEntity savedEntity;
+		Optional<BatchEntity> entityOpt = batchRepository.findById(batchDbId);
+		if (entityOpt.isPresent()) {
+			BatchEntity entity = entityOpt.get();
+			updateEntity(entity, batch);
+			entity.setDateModified(new Date());
+
+			savedEntity = batchRepository.save(entity);
+		} else {
+			throw new BrAPIServerDbIdNotFoundException("batch", batchDbId, HttpStatus.NOT_FOUND);
+		}
+
+		return convertToDetails(savedEntity);
+	}
+
+	public List<BatchSummary> saveNewBatch(@Valid List<BatchNewRequest> requests) throws BrAPIServerException {
+
+		List<BatchSummary> savedBatches = new ArrayList<>();
+
+		for (BatchNewRequest batch : requests) {
+
+			BatchEntity entity = new BatchEntity();
+			updateEntity(entity, batch);
+			Date now = new Date();
+			entity.setDateCreated(now);
+			entity.setDateModified(now);
+
+			BatchEntity savedEntity = batchRepository.save(entity);
+
+			savedBatches.add(convertToSummary(savedEntity));
+		}
+
+		return savedBatches;
+	}
+
+	public void deleteBatch(String batchDbId) throws BrAPIServerException {
+		BatchDetails batch = getBatch(batchDbId);
+		batchRepository.deleteAllByIdInBatch(batch.getData());
+	}
+
+	private BatchDetails convertToDetails(BatchEntity entity) {
+		BatchDetails details = new BatchDetails();
+		details = (BatchDetails) convertToBaseFields(entity, details);
+		details.setBatchDbId(entity.getId());
+		details.setData(entity.getData().stream().map((e) -> {
```

Comment:
could simplify a little:
`details.setData(entity.getData().stream().map(e -> e.getItem())
.collect(Collectors.toList()));`

Reply:
Yep, done.

Reply:
Marking resolved - this was changed to suggestion